modules/stake/keeper/validator.go

```diff
@@ -407,3 +409,29 @@ func (k Keeper) UnbondAllMatureValidatorQueue(ctx sdk.Context) {
 		store.Delete(validatorTimesliceIterator.Key())
 	}
 }
+
+// get the set of all validators with no limits, used during genesis dump
+func (k Keeper) AllValidatorsIterator(store sdk.KVStore) sdk.Iterator {
+	return sdk.KVStorePrefixIterator(store, ValidatorsKey)
+}
+
+func (k Keeper) InitMetrics(store sdk.KVStore) {
+	iterator := k.AllValidatorsIterator(store)
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		addr := iterator.Key()[1:]
+		validator := types.MustUnmarshalValidator(k.cdc, addr, iterator.Value())
+		tokenPrecision := sdk.NewIntWithDecimal(1, 18)
+		bondedToken, err := strconv.ParseFloat(validator.GetTokens().QuoInt(tokenPrecision).String(), 64)
```

Comment:
Let's replace magic number 18 in `sdk.NewIntWithDecimal(1, 18)` with something like: `sdk.AttoUnit(sdk.NativeTokenName).Decimal`

A bit too long, maybe add some convenience methods into CoinType?

Reply:
Added `AttoPrecision` in `coin_type`.